source/common/config/rds_json.cc

```diff
@@ -161,10 +161,16 @@ void RdsJson::translateRoute(const Json::Object& json_route, envoy::api::v2::Rou
   }
   if (json_route.hasObject("path")) {
     if (json_route.hasObject("prefix")) {
-      throw EnvoyException("routes must specify either prefix or path");
+      throw EnvoyException("routes must specify one of prefix, path, or regex");
     }
     match->set_path(json_route.getString("path"));
   }
+  if (json_route.hasObject("regex")) {
+    if (json_route.hasObject("prefix") || json_route.hasObject("path")) {
+      throw EnvoyException("routes must specify one of prefix, path, or regex");
+    }
+    match->set_regex(json_route.getString("regex"));
+  }
```

Comment:
Only one is allowed right? So make this a 3-way xor of all 3 options and check that only one is enabled. See the implementation around weighted cluster, cluster header and cluster for reference (I think it's in upstream_impl.cc)

Reply:
Actually in this same file